lib/multiple_search_selection.dart

```diff
@@ -579,32 +586,36 @@ class _MultipleSearchSelectionState<T> extends State<MultipleSearchSelection<T>>
     pickedItems.addAll(widget.initialPickedItems ?? []);
   }
 
+  List<T> _searchAllItems(String query, { bool fuzzy = false }) {
+    final q = widget.caseSensitiveSearch ? query : query.toLowerCase();
+
+    return allItems.where((item) {
+        final i = widget.caseSensitiveSearch ? widget.fieldToCheck(item) : widget.fieldToCheck(item).toLowerCase();
+        bool matches = i.contains(q);
+
+        if (fuzzy) {
+          if (widget.fuzzySearch == FuzzySearch.jaro) {
+            matches |= getJaro(widget.fieldToCheck(item), query, caseSensitive: widget.caseSensitiveSearch) >= 0.8;
+          } else if (widget.fuzzySearch == FuzzySearch.levenshtein) {
+            matches |= getLevenshtein(widget.fieldToCheck(item), query, caseSensitive: widget.caseSensitiveSearch) <= 2;
+          }
+        }
+
+        return matches;
+    }) .toList();
+  }
+
   void _onRemoveItem(T item) {
     pickedItems.remove(item);
     allItems.add(item);
-    showedItems = allItems
-        .where(
-          (item) => widget.fieldToCheck.call(item).contains(
-                _textEditingController.text,
-              ),
-        )
-        .toList();
-    if (showedItems.isNotEmpty) {
-      if (widget.sortShowedItems) {
-        showedItems.sort(
-          (a, b) => widget.fieldToCheck(a).compareTo(
-                widget.fieldToCheck(b),
-              ),
-        );
-      }
-    }
     if (widget.sortShowedItems) {
       allItems.sort(
-        (a, b) => widget.fieldToCheck(a).compareTo(
-              widget.fieldToCheck(b),
-            ),
+            (a, b) => widget.fieldToCheck(a).compareTo(
+          widget.fieldToCheck(b),
+        ),
       );
     }
+    showedItems = _searchAllItems( _textEditingController.text);
```

Comment:
correct me if i'm wrong here, but if we sort `allItems` first, we shouldn't also have to sort `showedItems`

Reply:
No you are correct, since we are always showing results based on `allItems` if `allItems` is sorted then we are good to go. Good catch.